packages/frontend-shared/src/components/ResultCounts.vue

```diff
@@ -21,14 +25,15 @@
 <script lang="ts" setup>
 
 export interface ResultCountsProps {
-  totalPassed: number|string|null
-  totalFailed: number|string|null
-  totalPending: number|string|null
-  totalSkipped: number|string|null
+  totalPassed?: number|string|null|SpecDataAggregate
```

Comment:
```suggestion
  totalPassed?: number | string | null | SpecDataAggregate
```

Why are these optional but then below are marked with the non-null `!` assertion? If you are rendering this component, these should be defined no?

Reply:
The props are optional because we do not always pass in 4 values to `Resultcounts` e.g. in `DebugSpec` we only pass in `totalPassed`, `totalFailed`, and `totalPending` but in `RunResult.vue` we pass in all 4 values. Hence it is not always thee case that these will be defined.
In order to avoid setting `undefined` props there is a non null `!` assertion below in `Order_Map`

Reply:
I wonder if this component is too generic, it seems to receive too many different values. Also, if we have `?`, shouldn't we not need `null`?